runtime/integration-tests/src/connectors_gateway/routers/ethereum_xcm/router.rs

```diff
@@ -0,0 +1,231 @@
+//
+// This file is part of the Centrifuge chain project.
+// Centrifuge is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version (see http://www.gnu.org/licenses).
+// Centrifuge is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+use ::xcm::{
+	latest::{Junction, Junction::*, Junctions::*, MultiLocation, NetworkId},
+	prelude::{Parachain, X1, X2},
+	VersionedMultiLocation,
+};
+use cfg_primitives::{PoolId, TrancheId};
+use cfg_traits::connectors::OutboundQueue;
+use cfg_types::{
+	domain_address::Domain,
+	fixed_point::Rate,
+	tokens::{CrossChainTransferability, CurrencyId, CustomMetadata},
+};
+use connectors_gateway_routers::{
+	ethereum_xcm::EthereumXCMRouter, AxelarXCMRouter, DomainRouter, EVMChain, EVMDomain, FeeValues,
+	XCMRouter, XcmDomain, XcmTransactInfo,
+};
+use frame_support::{assert_noop, assert_ok};
+use hex::FromHex;
+use orml_traits::{asset_registry::AssetMetadata, MultiCurrency};
+use pallet_connectors::Message;
+use runtime_common::{xcm::general_key, xcm_fees::default_per_second};
+use sp_core::{bounded::BoundedVec, H160};
+use xcm_emulator::TestExt;
+
+use crate::{
+	chain::centrifuge::{
+		Balance, ConnectorsGateway, OrmlAssetRegistry, OrmlTokens, Runtime, RuntimeOrigin,
+	},
+	connectors_gateway::routers::ethereum_xcm::{
+		setup::{dollar, ALICE, BOB, CHARLIE, PARA_ID_MOONBEAM, TEST_DOMAIN},
+		test_net::{Development, Moonbeam, RelayChain, TestNet},
+	},
+	utils::accounts::Keyring,
+};
+
+#[test]
+fn submit() {
+	submit_test_fn(get_ethereum_xcm_router_fn());
+
+	submit_test_fn(get_axelar_xcm_router_fn());
+}
+
+fn submit_test_fn(router_creation_fn: RouterCreationFn) {
+	TestNet::reset();
+
+	Development::execute_with(|| {
+		setup(router_creation_fn);
+
+		let msg = Message::<Domain, PoolId, TrancheId, Balance, Rate>::Transfer {
+			currency: 0,
+			sender: ALICE.into(),
+			receiver: BOB.into(),
+			amount: 1_000u128,
+		};
+
+		assert_ok!(<ConnectorsGateway as OutboundQueue>::submit(
+			ALICE.into(),
+			TEST_DOMAIN,
+			msg.clone(),
+		));
+
+		assert_noop!(
+			<ConnectorsGateway as OutboundQueue>::submit(
+				ALICE.into(),
+				Domain::EVM(1285),
+				msg.clone(),
+			),
+			pallet_connectors_gateway::Error::<Runtime>::RouterNotFound,
+		);
+
+		assert_noop!(
+			<ConnectorsGateway as OutboundQueue>::submit(CHARLIE.into(), TEST_DOMAIN, msg),
+			pallet_xcm_transactor::Error::<Runtime>::UnableToWithdrawAsset,
+		);
+	});
+}
+
+type RouterCreationFn = Box<
+	dyn Fn(VersionedMultiLocation, CurrencyId, VersionedMultiLocation) -> DomainRouter<Runtime>,
+>;
+
+fn get_axelar_xcm_router_fn() -> RouterCreationFn {
```

Comment:
I am pretty confused about this structure. We have `connectors_gateway/routers/{axelar_evm, ethereum_xcm}`, where **in the latter** we have a `test_net` and `setup.rs` and then a `router.rs` that seems to be testing both routers.

Reply:
I think I'll have to revisit the whole structure of connectors integration tests mainly due to this. Was thinking that we should have a `connectors` dir where we have the connectors pallet tests + gateway tests + router tests, which would remove some of the duplicate code.

Initially, I was under the impression that most of the `old connectors` tests will change but then realized that won't be the case.

Reply:
@NunoAlexandre I moved all connectors related tests i.e. gateway, connectors pallet, routers into one folder. I think this new structure fits better than the initial approach. PTAL